design/backup_cancellation.md

```diff
@@ -0,0 +1,154 @@
+```go
+type backupCancellationReconciler struct {
+    client.Client
+    logger                logrus.FieldLogger
+    itemOperationsMap     *itemoperationmap.BackupItemOperationsMap
+    newPluginManager      func(logger logrus.FieldLogger) clientmgmt.Manager
+    backupStoreGetter     persistence.ObjectBackupStoreGetter
+}
+```
+
+The controller will:
+1. Watch for backups in `BackupPhaseCancelling`
+2. Get operations from `itemOperationsMap.GetOperationsForBackup()`
+3. Call `bia.Cancel(operationID, backup)` on all in-progress BackupItemAction operations
+4. Find and cancel PodVolumeBackups by setting `pvb.Spec.Cancel = true`
+5. Wait for all cancellations to complete
+6. Set backup phase to `BackupPhaseCancelled`
+7. Update backup metadata in object storage
+
+### Cancellation Flow
+
+#### BackupItemAction Operations
+For operations with BackupItemAction v2 implementations (e.g., CSI PVC actions):
+1. Controller calls `bia.Cancel(operationID, backup)`
+2. CSI PVC action finds associated DataUpload and sets `du.Spec.Cancel = true`
+3. Node-agent DataUpload controller handles actual cancellation
+4. Operation marked as `OperationPhaseCanceled`
+
+#### PodVolumeBackup Operations
+For PodVolumeBackups (which lack BackupItemAction implementations):
+1. Controller directly finds PVBs by backup UID label
+2. Sets `pvb.Spec.Cancel = true` on in-progress PVBs
+3. Node-agent PodVolumeBackup controller handles actual cancellation
+
+
+## Alternatives Considered
+
+
+### Alternative 1: Deletion-Based Cancellation
+Using backup deletion as the cancellation mechanism instead of a cancel field.
+This was rejected because it doesn't allow users to preserve the backup object for inspection after cancellation, and deletion has different semantic meaning.
+
+### Alternative 2: Timeout-Based Automatic Cancellation
+Automatically cancelling backups after a configurable timeout.
+This was considered out of scope for the initial implementation as it addresses a different use case than user-initiated cancellation.
+
+## Security Considerations
+The cancel field requires the same RBAC permissions as updating other backup specification fields.
+No additional security considerations are introduced as the cancellation mechanism reuses existing operation cancellation pathways that are already secured.
+
+## Compatibility
+The new `cancel` field is optional and defaults to nil/false, ensuring backward compatibility with existing backup specifications.
+Existing backups will continue to work without modification.
+The new backup phases (`Cancelling`, `Cancelled`) are additive and don't affect existing phase transitions.
+
+## Implementation
+Implementation will be done incrementally in the following phases:
+
+**Phase 1**: API changes and basic cancellation detection
+- Add `cancel` field to BackupSpec
+- Add new backup phases
+- Update existing controllers to detect cancellation and transition to `Cancelling` phase
+
+**Phase 2**: Cancellation controller implementation
```

Comment:
Note that between Phase 1 and phase 2, "Cancelling" is essentially a terminal phase. Since we don't plan on having a release with only phase 1 in it, this should be fine.